fixtures/expiration/index.html

```diff
@@ -0,0 +1,57 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <title>Expiration Example</title>
+</head>
+
+<body>
+  <h1>Expiration Example</h1>
+  <p>Expected behavior: A counter that increases approximately once per second. If there's no counter, or if it doesn't increment
+    regularly, something is broken.</p>
+  <div id="container"></div>
+  <script src="../../build/dist/react.development.js"></script>
+  <script src="../../build/dist/react-dom.development.js"></script>
+  <script src="https://unpkg.com/babel-standalone@6/babel.js"></script>
+  <script type="text/babel">
+
+    // Clogs the main thread with synchronous work. The effect is that callbacks
+    // scheduled with requestIdleCallback will only fire if a `timeout`
+    // is specified.
+    function clogMainThread(controller, interval) {
+      function block() {
+        const fps = 60;
+        // Block the main thread for the rest of the frame. (Assumes 60fps.)
+        const e = performance.now() + (1000 / fps);
+        while (performance.now() < e) {}
+        requestAnimationFrame(block);
+      }
+      requestAnimationFrame(block);
+    }
+    clogMainThread();
+
+    class Counter extends React.unstable_AsyncComponent {
+      state = {counter: 0};
+      increment = () => {
+        this.setState(state => ({counter: state.counter + 1}));
+      }
+      componentDidMount() {
+        setTimeout(this.increment, 0);                
```

Comment:
The timeout is a workaround until we have a way to schedule an async update from inside a lifecycle (passive `componentDidUpdate`).